### PR TITLE
SEDIT-TOPLEVEL:  Add functions for setting/getting properties #604

### PR DESCRIPTION
**Pull request by rmkaplan**
also a field in the edit-context to store them, and a change to the region subroutine to respect the :REGION property.

Note that SEDIT has a peculiarity in that the record declarations are on SEDIT-DECLS but the runtime INITRECORDS is separately on SEDIT-ACCESS.  So you have to make sure you dump SEDIT-ACCESS, not just SEDIT-DECLS if a record changes.  A bit of a trap

### CODE REVIEW

File: sources/SEDIT-TOPLEVEL [hunk 1]
```diff
@@ -231,8 +265,7 @@
 
               (IL:* IL:|;;| "found a matching context elsewhere")
 
-              (IL:RESETFORM (IL:EDITMODE SEDIT)
-                     (IL:EDITDEF NAME TYPE NIL NIL '(:DONTWAIT))))))))
+              (IL:EDITDEF NAME TYPE NIL NIL '(:DONTWAIT)))))))
```
Comment (masinter) [inline sources/SEDIT-TOPLEVEL:269]: it looks like this PR loses the edit I made in a previous PR

**Review by masinter — CHANGES_REQUESTED** [review]

**Comment by rmkaplan** [comment]
Added your MARKASCHANGEDFN fix.  Might have missed it before because of a delayed merge

File: sources/SEDIT-TOPLEVEL (hunk 1 above)
Comment (masinter) [reply sources/SEDIT-TOPLEVEL]: this is screwy -- I updated rmk6-redux to have the right EDITMODE but it should have been this one rmk7 instead.
once more? 

i got an error that I haven't tracked down -- too many arguments (2) to sedit::start-process (which only takes one).

**Comment by rmkaplan** [comment]
I think the missing quote explains an error that I got, of a NOBIND when I was just trying to do a COPYDEF.  It got the error inside FNS.PUTDEF.

**Comment by rmkaplan** [comment]
Should I put the quote in my version (which has the other stuff), and then commit it back to rmk7?

File: sources/SEDIT-TOPLEVEL [hunk 2]
```diff
@@ -119,20 +121,38 @@
...
+
+    (IL:* IL:|;;| 
+    "Edited  4-Dec-2021 21:54 by rmk: The :REGION property gives the user directe control")
+
+    (IL:* IL:|;;| "RMK: The :REGION property gives the user directe control")
+                                                     (IL:* IL:\; "Edited  1-Dec-2021 22:51 by rmk:")
+                                                             (IL:* IL:\; 
+                                                             "Edited 19-Nov-87 10:18 by DCB")
+
+(IL:* IL:|;;;| "called to get a region for this sedit window.  should return the region for the sedit including the prompt window.  context is being built and needs a window.  the context will have at least the name (IconTitle) and type (EditType) of the object being edited, and can be used as desired to map between contexts and windows.  If reason is :CREATE, then this function must return a region.  If :EXPAND, then this algorithm returns a region from the stack only if SEDIT.KEEP.WINDOW.REGION is nil, otherwise it returns NIL, telling the window system not to reshape on expansion.")
```
Comment (masinter) [inline sources/SEDIT-TOPLEVEL:135]: comment about the REGION property appears twice

**Review by masinter — APPROVED** [review]

**Pull request closed without merging**